packages/main/src/Carousel.js

```diff
@@ -0,0 +1,296 @@
+		 * @type {Boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		cycling: {
+			type: Boolean,
+		},
+
+		/**
+		 * Sets the amount of items per page. If this property is set, on mobile devices it will always fallback to 1.
+		 * @type {Integer}
+		 * @defaultvalue 1
+		 * @public
+		 */
+		itemsPerPage: {
+			type: Integer,
+			defaultValue: 1,
+		},
+
+		/**
+		 * If set to true the navigation is hidden.
+		 * @type {Boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		hideNavigation: {
+			type: Boolean,
+		},
+
+		/**
+		 * Defines the index of the initially selected slot.
+		 * @type {Integer}
+		 * @defaultvalue 0
+		 * @public
+		 */
+		selectedIndex: {
+			type: Integer,
+			defaultValue: 0,
+		},
+
+		/**
+		 * Defines the position of arrows.
+		 * @type {CarouselArrowsPlacement}
+		 * @defaultvalue CarouselArrowsPlacement.Content
+		 * @public
+		 */
+		arrowsPlacement: {
+			type: CarouselArrowsPlacement,
+			defaultValue: CarouselArrowsPlacement.Content,
+		},
+	},
+	slots: /** @lends sap.ui.webcomponents.main.Carousel.prototype */ {
+		/**
+		 * Defines the content of the <code>ui5-carousel</code>.
+		 * @type {HTMLElement[]}
+		 * @slot
+		 * @public
+		 */
+		"default": {
+			propertyName: "content",
+			type: HTMLElement,
+			individualSlots: true,
+		},
+	},
+	events: /** @lends sap.ui.webcomponents.main.Carousel.prototype */ {
+		//
+	},
+};
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ * The carousel allows the user to browse through a set of items by swiping right or left.
+ * The component is mostly used for showing a gallery of images, but can hold any other HTML element.
+ *
+ * <h3>Usage</h3>
+ *
+ * When to use
+ *   - The items you want to display are very different from each other.
+ *   - You want to display the items one after the other.
+ * When not to use
+ *   - The items you want to display need to be visible at the same time.
+ *   - The items you want to display are uniform and very similar
+ *
+ * For the <code>ui5-carousel</code>
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents/dist/Carousel.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.Carousel
+ * @extends UI5Element
+ * @tagname ui5-carousel
+ * @since 1.0.0-rc.6
+ * @public
+ */
+class Carousel extends UI5Element {
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return CarouselCss;
+	}
+
+	static get template() {
+		return CarouselTemplate;
+	}
+
+	static get pageTypeLimit() {
+		return 9;
+	}
+
+	constructor() {
+		super();
+
+		this._scrollEnablement = new ScrollEnablement(this);
+		this._scrollEnablement.attachEvent("touchend", event => {
+			this._updateScrolling(event);
+		});
+
+		this.i18nBundle = getI18nBundle("@ui5/webcomponents");
+	}
+
+	onBeforeRendering() {
+		if (!isDesktop()) {
+			this.itemsPerPage = 1;
```

Comment:
We should never mutate the public APIs (except when there is user input). Better create a getter:
get effectiveItemsPerPage() {
  return mobile ? 1 : this.itemsPerPage
}